### PR TITLE
Experimental filters: use context for data providers

### PR DESCRIPTION
**Pull request by krzysztofzuraw**
This PR extracts providers into context so we can inject query params into the `ProductList` view and fetch a list of products with selected filters.

ref: #3272

### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `data-test-id` are added for new elements
4. [ ] The changes are tested in Chrome/Firefox/Safari browsers and in light/dark mode
5. [ ] Your code works with the latest stable version of the core
6. [x] I added changesets and [read good practices](/.changeset/README.md)

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [ ] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants
23. [ ] payments

CONTAINERS=1

### CODE REVIEW

**Review by Droniu — APPROVED** [review]

File: src/components/ConditionalFilter/index.tsx [hunk 1]
```diff
@@ -1,24 +1,11 @@
...
 
 export const ConditionalProductFilters = () => {
-  const provider = useUrlValueProvider();
-  const apiProvider = useProductFilterAPIProvider();
-  const leftOperandsProvider = useFilterLeftOperandsProvider();
-
   // @ts-expect-error
-  const handleConfirm = (value: FilterContainer) => {
-  }
+  const handleConfirm = (value: FilterContainer) => {};
```
Comment (poulch) [inline src/components/ConditionalFilter/index.tsx:8]: question: do you need this?

File: src/components/ConditionalFilter/ConditionalFilters.tsx [hunk 2]
```diff
@@ -1,37 +1,37 @@
...
-    onConfirm(value)
-  }
+    valueProvider.persist(value);
+    // valueProvider.navigate(value);
+    onConfirm(value);
+  };
+
+  // const handleClear = () => {
+  // valueProvider.clear();
+  // }
```
Comment (poulch) [inline src/components/ConditionalFilter/ConditionalFilters.tsx:23]:
```suggestion
```

File: src/components/ConditionalFilter/ConditionalFilters.tsx [hunk 3]
```diff
@@ -1,37 +1,37 @@
...
-export const ConditionalFilters = ({ valueProvider, apiProvider, leftOperandsProvider, onConfirm }: ConditionalFiltersProps) => {
+export const ConditionalFilters = ({ onConfirm }: ConditionalFiltersProps) => {
+  const { valueProvider } = useConditionalFilterContext();
+
   const handleConfirm = (value: FilterContainer) => {
-    valueProvider.persist(value)
-    onConfirm(value)
-  }
+    valueProvider.persist(value);
+    // valueProvider.navigate(value);
```
Comment (poulch) [inline src/components/ConditionalFilter/ConditionalFilters.tsx:17]:
```suggestion
```

File: src/components/ConditionalFilter/API/ProductFilterAPIProvider.tsx [hunk 4]
```diff
@@ -79,8 +85,31 @@ export const useProductFilterAPIProvider = (): FilterAPIProvider => {
     return handler.fetch();
   };
 
+  const useInitialState = (fetchingParams: FetchingParams) => {
```
Comment (poulch) [inline src/components/ConditionalFilter/API/ProductFilterAPIProvider.tsx:88]: suggestion: `useInitialState` sounds like hook but this seems to be more like normal method/function

**Review by poulch — APPROVED** [review]

File: src/components/ConditionalFilter/API/ProductFilterAPIProvider.tsx (hunk 4 above)
Comment (andrzejewsky) [reply src/components/ConditionalFilter/API/ProductFilterAPIProvider.tsx:88]: it calls hook inside (useDataFromAPI) which calls react hooks

**Review by andrzejewsky — APPROVED** [review]

**Review by andrzejewsky — APPROVED** [review]

**Review by poulch — APPROVED** [review]

**Pull request merged**